src/Main.java

```diff
@@ -1,5 +1,91 @@
+import java.util.ArrayList;
+import java.util.List;
+
 public class Main {
     public static void main(String[] args) {
-        System.out.println("Hello world!");
+        Main main = new Main();
+        System.out.println("-------------------1-----------------");
+        System.out.println(main.maxDuplicate(List.of(1, 2, 3)));
+        System.out.println(main.maxDuplicate(List.of(1, 2, 2)));
+        System.out.println(main.maxDuplicate(List.of(1, 2, 2, 3, 1)));
+        System.out.println("-------------------2-----------------");
+        System.out.println(main.wordNumeration(new ArrayList<>(List.of("hi", "hi", "welcome"))));
+        System.out.println(main.wordNumeration(new ArrayList<>(List.of("Hi", "hi", "welcome", "no", "hI", "HI"))));
+        System.out.println("-------------------3-----------------");
+        System.out.println(main.makeBricks(3, 1, 8) );
+        System.out.println(main.makeBricks(3, 1, 9) );
+        System.out.println(main.makeBricks(3, 2, 10) );
+    }
+
+    /**
+     * Первый вариант решения
+     * public Integer maxDuplicate(List<Integer> nums) {
+     * List<Integer> num = new ArrayList<>();
+     * int count = 1;
+     * for (int i = 0; i < nums.size(); i++) {
+     * for (int y = i + 1; y < nums.size(); y++) {
+     * if (nums.get(i).equals(nums.get(y))) {
+     * count++;
+     * }
+     * if (count > 1) {
+     * num.add(nums.get(i));
+     * count = 1;
+     * }
+     * <p>
+     * }
+     * }
+     * if (num.isEmpty()) {
+     * return null;
+     * }
+     * return Collections.max(num);
+     * }
+     **/
+
+    public Integer maxDuplicate(List<Integer> nums) {
+        Integer a = 0;
+        int count = 1;
+        for (int i = 0; i < nums.size(); i++) {
+            for (int y = i + 1; y < nums.size(); y++) {
+                if (nums.get(i).equals(nums.get(y)) && nums.get(i) > a) {
+                    count++;
+                    a = nums.get(i);
+                }
+            }
+        }
+        if (count == 1) {
+            return null;
+        }
+        return a;
+    }
```

Comment:
This is a working solution, however, the complexity of it is O(n^2) because we have nested loops. For each element, we are iterating over all the remaining elements.
You don't have to change it but maybe you can try to make it more efficient, if you have time.
Tip: Try to use Set and only one loop